js/wholepage.js

```diff
@@ -1,19 +1,65 @@
 chrome.extension.onMessage.addListener(
-  function (request, sender, sendResponse) {
-      if (request.msg == "get_content") {
-          xmlhttp = new XMLHttpRequest();
-          xmlhttp.onreadystatechange = function () {
-              if (xmlhttp.readyState == 4) {
-                  if (xmlhttp.status == 200) {
-                      document.getElementsByTagName('html')[0].innerHTML = xmlhttp.responseText;
-                  } else {
-                      alert('Cannot reach russiangram.com');
-                  }
-              }
-          }
-          xmlhttp.open("POST", "https://russiangram.com/translate/Default.aspx", true);
-          xmlhttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
-          xmlhttp.send(document.getElementsByTagName('html')[0].innerHTML);
-          sendResponse({ data: 'success' });
-      }
-  });
+    function (request, sender, sendResponse) {
+        if (request.msg == "get_content") {
+            var tagList = document.getElementsByTagName('*');
+
+            var xmlhttp = new XMLHttpRequest();
+            xmlhttp.onreadystatechange = function () {
+                if (xmlhttp.readyState === 4) {
+                    if (xmlhttp.status === 200) {
+                        var data = JSON.parse(xmlhttp.responseText);
+                        updateText(data, tagList);
+                    } else {
+                        alert('Cannot reach russiangram.com');
+                    }
+                }
+            }
+
+            var collectedText = collectText(tagList);
+            var data = JSON.stringify(collectedText);
+
+            xmlhttp.open("POST", "https://russiangram.com/translate/Default.aspx", true);
+            xmlhttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+            xmlhttp.send(data);
+            sendResponse({data: 'success'});
+        }
+    });
+
+function collectText(tagList) {
+
+    var collectedText = {};
+
+    forEachTextNode(tagList, function (node, uid) {
+        collectedText[uid] = node.nodeValue;
+        node.uid = uid;
+    });
+
+    return collectedText;
+}
+
+function updateText(collectedText, tagList) {
+
+    forEachTextNode(tagList, function (node, uid) {
+        if (node.uid !== uid) {
+            alert('Oops, something went wrong. Please Try Again.\n\nThe RussianGram.com team.');
+            throw new Error('While waiting for a response from the server, the DOM was changed.');
```

Comment:
On some websites this might be very common. I suggest updating whatever nodes we can and then displaying one message at the end saying "Some parts of the page have not been stress marked. Please try again."